packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -650,21 +637,18 @@ export class SDKClient {
 
       throw sdkClientError;
     } finally {
-      /**
-       * @note Capturing the charged transaction fees at the end of the flow ensures these fees are eventually
-       *       captured in the metrics and rate limiter class, even if SDK transactions fail at any point.
-       */
+      // emitting an EXECUTE_QUERY event to kick off capturing metrics process asynchronously
       if (queryCost && queryCost !== 0) {
-        this.addExpenseAndCaptureMetrics(
-          `TransactionExecution`,
-          query.paymentTransactionId!,
-          queryType,
+        this.eventEmitter.emit(constants.EVENTS.EXECUTE_QUERY, {
+          executionType: `TransactionExecution`,
```

Comment:
We are executing a query but here we set the execution type to `TransactionExecution`? Seems odd.

Reply:
Ha good point! Probably copy and paste issue.